WhisperAPI/WhisperAPI.Tests/Unit/IndexSearchTest.cs

```diff
@@ -0,0 +1,86 @@
+                }));
+
+            this._httpClient = new HttpClient(httpMessageHandler.Object);
+
+            this._indexSearch = new IndexSearch(null, this._httpClient);
+
+            this._searchResult = new SearchResult
+            {
+                NbrElements = 4,
+                Elements = new List<SearchResultElement>
+                {
+                    new SearchResultElement
+                    {
+                        Title = "Available Coveo Cloud V2 Source Types",
+                        Uri = "https://onlinehelp.coveo.com/en/cloud/Available_Coveo_Cloud_V2_Source_Types.htm",
+                        PrintableUri = "https://onlinehelp.coveo.com/en/cloud/Available_Coveo_Cloud_V2_Source_Types.htm",
+                        Summary = null,
+                        Score = 4280
+                    },
+                    new SearchResultElement
+                    {
+                        Title = "Coveo Cloud Query Syntax Reference",
+                        Uri = "https://onlinehelp.coveo.com/en/cloud/Coveo_Cloud_Query_Syntax_Reference.htm",
+                        PrintableUri = "https://onlinehelp.coveo.com/en/cloud/Coveo_Cloud_Query_Syntax_Reference.htm",
+                        Summary = null,
+                        Score = 3900
+                    },
+                    new SearchResultElement
+                    {
+                        Title = "Events",
+                        Uri = "https://developers.coveo.com/display/JsSearchV1/Page/27230520/27230472/27230573",
+                        PrintableUri = "https://developers.coveo.com/display/JsSearchV1/Page/27230520/27230472/27230573",
+                        Summary = null,
+                        Score = 2947
+                    },
+                    new SearchResultElement
+                    {
+                        Title = "Coveo Facet Component (CoveoFacet)",
+                        Uri = "https://coveo.github.io/search-ui/components/facet.html",
+                        PrintableUri = "https://coveo.github.io/search-ui/components/facet.html",
+                        Summary = null,
+                        Score = 2932
+                    }
+                }
+            };
+        }
+
+        [Test]
+        [TestCase("test")]
+        public void Receive_response_from_post_then_return_result(string querry)
+        {
+            this._indexSearch.Search(querry).Should().BeEquivalentTo(this._searchResult);
```

Comment:
Dans un test, tu as 3 grandes sections, la phase initialisation où tu créer des objets que tu as besoin, tu fais tes Verifiy, Setup.Returns. Ensuite la phase de call qui serait this._indexSearch.Search() et finalement la phase de vérification qui est le reste de ta ligne